test/jdk/sun/awt/image/OffScreenImageSource/bug4200096.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import java.awt.image.ColorModel;
+import java.awt.image.ImageConsumer;
+import java.awt.image.ImageObserver;
+import java.io.PrintStream;
+import java.util.Hashtable;
+
+/**
+ * @test
+ * @bug 4200096
+ * @summary OffScreenImageSource.removeConsumer can cause NullPointerException
+ * @author Jeremy Wood
+ */
+
+public class bug4200096 {
+    public static void main(String[] args) {
+        // OffScreenImageSourceTest wraps image production in a try/catch.
+        // It catches NullPointerExceptions and prints them to System.err.
+        // So here we override System.err to intercept them and fail this test:
+        System.setErr(new PrintStream(System.err) {
+            @Override
+            public void println(Object x) {
+                super.println(x);
+                if (x instanceof Throwable)
+                    System.exit(1);
```

Comment:
The tests should not use the System.exit() as it might affect the execution of other tests. It is better to set some flag and check it at the end.

Reply:
Thanks; this is fixed.